iquip/apps/logger.py

```diff
@@ -125,52 +137,75 @@ class LoggerApp(qiwis.BaseApp):
     Attributes:
         loggerFrame: A frame that shows the logs.
         confirmFrame: A frame that asks whether to clear logs.
-        handler: A handler for adding logs to the loggerFrame. 
+        levelsDict: A dictionary that matches int level to string level.
+        frameHandler: A handler for adding logs to the loggerFrame.
+        fileHandler: A handler for saving logs to file.
     """
 
-    def __init__(self, name: str, parent: Optional[QObject] = None):
-        """Extended."""
+    def __init__(self, name: str, path: str, parent: Optional[QObject] = None):
+        """Extended.
+        
+        Args:
+            name: Specifies name of the LoggerApp.
+            path: Desribes file directory of log record file.
+        """
         super().__init__(name, parent=parent)
         self.loggerFrame = LoggerFrame()
         self.confirmFrame = ConfirmClearingFrame()
         # connect signals to slots
         self.loggerFrame.clearButton.clicked.connect(self.checkToClear)
         self.confirmFrame.confirmed.connect(self.clearLog)
-        self.handler = LoggingHandler(self.addLog)
-        # TODO(aijuh): Change the log format when it is determined.
-        fs = "%(levelname)s [%(name)s] [%(filename)s:%(lineno)d] %(message)s"
-        formatter = logging.Formatter(fs)
-        self.handler.setFormatter(formatter)
+        # initialize handlers
+        self.levelsDict = {logging.DEBUG: "DEBUG", logging.INFO: "INFO", logging.WARNING: "WARNING",
+                           logging.ERROR: "ERROR", logging.CRITICAL: "CRITICAL"}
+        self.frameHandler = LoggingHandler(self.addLog)
+        logFileName = os.path.join(path, QDateTime.currentDateTime().toString("logyyMMdd-HHmmss"))
+        self.fileHandler = handlers.TimedRotatingFileHandler(filename=logFileName, when="midnight",
+                                                             interval=1, encoding="utf-8")
+        self.initLogger()
+        # set loggerFrame's frameLevelBox
+        frameLevelBox = self.loggerFrame.frameLevelBox
+        frameLevelBox.addItems(self.levelsDict.values())
+        frameLevelBox.textActivated.connect(partial(self.setLevel, self.frameHandler))
+        frameLevelBox.setCurrentText(self.levelsDict[self.frameHandler.level])
+        # set loggerFrame's fileLevelBox
+        fileLevelBox = self.loggerFrame.fileLevelBox
+        fileLevelBox.addItems(self.levelsDict.values())
+        fileLevelBox.textActivated.connect(partial(self.setLevel, self.fileHandler))
+        fileLevelBox.setCurrentText(self.levelsDict[self.fileHandler.level])
+
+    def initLogger(self):
```

Comment:
If you use `initLogger` method, then it looks better to move the parts initializing loggers in the constructor into this method.

Reply:
I requested to define all the attributes in `__init__()`.
@Aijuh You can discuss why you did it that way or ask me about this.

In addition, "initialize" and "instantiate" are different.

Reply:
I separated constructor into two parts because constructor looked to long. 
@kangz12345 is it okay to move all the codes into constructor?

In addition, since log and handler class became an specific object, instantiate was right in this case.